InnerEye/ML/reports/classification_report.py

```diff
@@ -176,92 +314,138 @@ def get_metric(val_labels_and_predictions: LabelsAndPredictions,
                                              config: ScalarModelBase,
-                                             is_thresholded: bool = False) -> None:
+                                             val_data_split: Optional[ModelExecutionMode] = None,
+                                             test_data_split: Optional[ModelExecutionMode] = None,
+                                             is_thresholded: bool = False,
+                                             is_crossval_report: bool = False) -> None:
     """
     Given csvs written during inference for the validation and test sets, print out metrics for every prediction target
     in the config.
 
     :param val_metrics_csv: Csv written during inference time for the val set. This is used to determine the
-    optimal threshold for classification. 
+    optimal threshold for classification.
     :param test_metrics_csv: Csv written during inference time for the test set. Metrics are calculated for this csv.
     :param config: Model config
+    :param val_data_split: Whether to filter the validation CSV file for Train, Val, or Test results (default: no
+    filtering).
+    :param test_data_split: Whether to filter the test CSV file for Train, Val, or Test results (default: no filtering).
     :param is_thresholded: Whether the model outputs are binary (they have been thresholded at some point)
                            or are floating point numbers.
-    """
+    :param is_crossval_report: If True, assumes CSVs contain results for multiple cross-validation runs and prints the
+    metrics along with means and standard deviations. Otherwise, prints metrics for a single run.
+    """
+    def get_metrics_for_fold(prediction_target: str, crossval_split: Optional[int] = None) -> Dict[str, float]:
+        val_labels_and_predictions = get_labels_and_predictions(val_metrics_csv, prediction_target,
+                                                                crossval_split_index=crossval_split,
+                                                                data_split=val_data_split)
+        test_labels_and_predictions = get_labels_and_predictions(test_metrics_csv, prediction_target,
+                                                                 crossval_split_index=crossval_split,
+                                                                 data_split=test_data_split)
+        return get_all_metrics(val_labels_and_predictions, test_labels_and_predictions, is_thresholded)
 
     for prediction_target in config.class_names:
-        print_header(f"Class {prediction_target}", level=3)
-        val_metrics = get_labels_and_predictions(val_metrics_csv, prediction_target)
-        test_metrics = get_labels_and_predictions(test_metrics_csv, prediction_target)
-        print_metrics(val_labels_and_predictions=val_metrics, test_labels_and_predictions=test_metrics, is_thresholded=is_thresholded)
+        print_header(f"Class: {prediction_target}", level=3)
```

Comment:
This is quite hard to parse, would you mind adding a few comments telling what each block is doing?

Reply:
sure, done